### PR TITLE
COMMERCE-12144 Min quantity can be anything greater than 0

### PR DESCRIPTION
**Pull request by dsitu**
https://liferay.atlassian.net/browse/COMMERCE-12144

### CODE REVIEW

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/EditCPDefinitionMVCActionCommand.java [hunk 1]
```diff
@@ -598,11 +598,11 @@ private void _updateCPDefinitionInventory(
 		boolean backOrders = ParamUtil.getBoolean(actionRequest, "backOrders");
 		int minStockQuantity = ParamUtil.getInteger(
 			actionRequest, "minStockQuantity");
-		int minOrderQuantity = ParamUtil.getInteger(
+		double minOrderQuantity = ParamUtil.getDouble(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/EditCPDefinitionMVCActionCommand.java:601]: change this to BigDecimal

File: modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_cp_definition_configuration.jsp [hunk 2]
```diff
@@ -107,10 +107,10 @@ boolean shippable = BeanParamUtil.getBoolean(cpDefinition, request, "shippable",
 						}
 						%>
 
-						<aui:input ignoreRequestValue="<%= true %>" name="minOrderQuantity" type="text" value="<%= String.valueOf(minOrderQuantity.intValue()) %>">
-							<aui:validator name="digits" />
+						<aui:input ignoreRequestValue="<%= true %>" name="minOrderQuantity" type="text" value="<%= String.valueOf(minOrderQuantity.doubleValue()) %>">
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_cp_definition_configuration.jsp:110]: please have a look to some price page to use bigdecimal also here

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/rest-openapi.yaml [hunk 3]
```diff
@@ -279,8 +279,8 @@ components:
                     readOnly: true
                     type: object
                 quantity:
-                    format: int32
-                    type: integer
+                    format: double
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/rest-openapi.yaml:282]: sync with commits from COMMERCE-11747

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Don't use Double, but BigDecimal

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/rest-openapi.yaml (hunk 3 above)
Comment (dsitu) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/rest-openapi.yaml]: talked to @smottal, confirmed that his work on https://liferay.atlassian.net/browse/COMMERCE-11747, will include headless changes for updating quantity related fields to BigDecimal.

**Comment by dsitu** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
LGTM, but there are some conflicts to solve before I can move this to QA check.

Thank you!

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
@andrea-ale-sbarra conflicts resolved

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by 22LucaGiannotta22** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
Resent here: https://github.com/liferay-commerce/liferay-portal/pull/4189

**Pull request closed without merging**